Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -0,0 +1,828 @@
+  int nStepsInsideSameStrip[2] = { 0, 0 }; // number of propagation steps in the same strip (since we have maximum 2 strips, it has dimention = 2)
+  float deltaPosTemp[3];
+  std::array<float, 3> pos;
+  std::array<float, 3> posBeforeProp;
+  float posFloat[3];
+
+  LOG(DEBUG) << "Trying to match %d tracks" << cacheTrk.size();
+  for (int itrk = 0; itrk < cacheTrk.size(); itrk++) {
+    for (int ii = 0; ii < 2; ii++) {
+      detId[ii][2] = -1; // before trying to match, we need to inizialize the detId corresponding to the strip number to -1; this is the array that we will use to save the det id of the maximum 2 strips matched
+      nStepsInsideSameStrip[ii] = 0;
+    }
+    int nStripsCrossedInPropagation = 0; // how many strips were hit during the propagation
+    auto& trefTrk = mTracksWork[cacheTrk[itrk]];
+    float minTrkTime = (trefTrk.getTimeMUS().getTimeStamp() - mSigmaTimeCut * trefTrk.getTimeMUS().getTimeStampError()) * 1.E6; // minimum time in ps
+    float maxTrkTime = (trefTrk.getTimeMUS().getTimeStamp() + mSigmaTimeCut * trefTrk.getTimeMUS().getTimeStampError()) * 1.E6; // maximum time in ps
+    int istep = 1;                                                                                                              // number of steps
+    float step = 0.1;                                                                                                           // step size in cm
+                                                                                                                                //uncomment for local debug
+                                                                                                                                /*
+																//trefTrk.getXYZGlo(posBeforeProp);
+																//float posBeforeProp[3] = {trefTrk.getX(), trefTrk.getY(), trefTrk.getZ()}; // in local ref system
+																//printf("Global coordinates: posBeforeProp[0] = %f, posBeforeProp[1] = %f, posBeforeProp[2] = %f\n", posBeforeProp[0], posBeforeProp[1], posBeforeProp[2]);
+																//Printf("Radius xy = %f", TMath::Sqrt(posBeforeProp[0]*posBeforeProp[0] + posBeforeProp[1]*posBeforeProp[1]));
+																//Printf("Radius xyz = %f", TMath::Sqrt(posBeforeProp[0]*posBeforeProp[0] + posBeforeProp[1]*posBeforeProp[1] + posBeforeProp[2]*posBeforeProp[2]));
+																*/
+
+#ifdef _ALLOW_DEBUG_TREES_
+    (*mDBGOut) << "propOK"
+               << "track=" << trefTrk << "\n";
+#endif
+
+    // initializing
+    for (int ii = 0; ii < 2; ii++) {
+      for (int iii = 0; iii < 5; iii++) {
+        detId[ii][iii] = -1;
+      }
+    }
+    int detIdTemp[5] = { -1, -1, -1, -1, -1 }; // TOF detector id at the current propagation point
+    while (propagateToRefX(trefTrk, mXRef + istep * step, step) && nStripsCrossedInPropagation <= 2 && mXRef + istep * step < Geo::RMAX) {
+      if (0 && istep % 100 == 0) {
```

Comment:
@chiarazampolli : Here also, I would not propagate from the full track in a 1mm steps. Instead, I would to smt. like:
```
TrackPar parOnly(trefTrk);
while (propagateToRefXNoCovariance( parOnly, mXRef + istep * step, step) ..., full field) {
    // check for strip crossing
    if ( .. crossed ) { // do full error 
         // do a few cm-size step full propagation to already found R.
    } 
}
}

Given that this part will certainly change once we start with outer track param, we can optimize it later.

Reply:
Hi @shahor02 , 

Yes, this is what I discussed with Francesco. Before changing it, we wanted to check with you if the way we did it in the first part was ok. There is no problem in changing here too, just let us know (in case you have a second thought :) ).

I have a question: suppose that we find a strip at 4.3 cm after 371 (just to say a number): when we say: "propagate to 371+4.3 in steps of 3 cm", will it do 1 step on 3 cm and 1 step of 1.3?

Chiara